handlers/jaeger.go

```diff
@@ -1,31 +1,76 @@
 package handlers
 
 import (
+	"errors"
 	"net/http"
+	"time"
 
 	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
 	"github.com/kiali/kiali/models"
+	"github.com/kiali/kiali/status"
+	"github.com/kiali/kiali/util/httputil"
 )
 
-// Get JaegerInfo provides the proxy Jaeger URL
+// Get JaegerInfo provides the Jaeger URL and other info, first by checking if a config exists
+// then (if not) by inspecting the Kubernetes Jaeger service in Istio installation namespace
 func GetJaegerInfo(w http.ResponseWriter, r *http.Request) {
-	jaegerConfig := config.Get().ExternalServices.Tracing
-	info := models.JaegerInfo{
-		URL: jaegerConfig.URL,
+	requestToken, err := getToken(r)
+	if err != nil {
+		RespondWithError(w, http.StatusInternalServerError, "Token initialization error: "+err.Error())
+		return
 	}
 
-	// Check if URL is in the configuration
-	if info.URL == "" {
-		RespondWithError(w, http.StatusNotFound, "You need to set the Jaeger URL configuration.")
+	info, code, err := getJaegerInfo(requestToken)
+	if err != nil {
+		log.Error(err)
+		RespondWithError(w, code, err.Error())
 		return
 	}
+	RespondWithJSON(w, code, info)
+}
+
+func getJaegerInfo(requestToken string) (*models.JaegerInfo, int, error) {
+	jaegerConfig := config.Get().ExternalServices.Tracing
+
+	if !jaegerConfig.Enabled {
+		return nil, http.StatusNoContent, nil
+	}
+
+	apiURL := status.DiscoverJaeger()
+	if apiURL == "" {
+		return nil, http.StatusServiceUnavailable, errors.New("Jaeger URL is not set in Kiali configuration")
+	}
 
 	// Check if URL is valid
-	_, error := validateURL(info.URL)
-	if error != nil {
-		RespondWithError(w, http.StatusNotAcceptable, "You need to set a correct format for Jaeger URL in the configuration error: "+error.Error())
-		return
+	_, err := validateURL(apiURL)
+	if err != nil {
+		return nil, http.StatusServiceUnavailable, errors.New("wrong format for Jaeger URL: " + err.Error())
 	}
 
-	RespondWithJSON(w, http.StatusOK, info)
+	// Be sure to copy config.Auth and not modify the existing
+	auth := jaegerConfig.Auth
+	if auth.UseKialiToken {
+		auth.Token = requestToken
+	}
+
+	ha, code, err := canAccessURL(apiURL, &auth)
+	if !ha {
+		return nil, code, errors.New("can't reach Jaeger query service")
```

Comment:
I would change this if by this because you are going to use this variables only in the if
```go
if ha, code, err := canAccessURL(apiURL, &auth);!ha {
		return nil, code, errors.New("can't reach Jaeger query service")
	} else if err != nil {
		return nil, http.StatusInternalServerError, err
	}
```